### ISSUE TITLE
training_v2 ignores --precision fp32 when selecting Fabric precision on MPS

### ISSUE THREAD

**Issue by denisroldan** [ace-step/ACE-Step-1.5#1244]
## Summary

`train_fixed --device mps --precision fp32` displays/loads fp32 but the training loop still selects MPS defaults (`torch.float16`, `16-mixed`), so the user cannot force true fp32 training from the CLI.

## Expected

`--precision fp32` should select `torch.float32`, Fabric `32-true`, and avoid autocast.

## Actual

`FixedLoRATrainer` calls `_select_compute_dtype(device_type)` and `_select_fabric_precision(device_type)` without passing `cfg.precision`, so MPS always resolves to fp16/`16-mixed`.

A focused local check against the current `HEAD` sources shows:

```text
HEAD _select_compute_dtype signature: (device_type: str) -> torch.dtype
HEAD _select_fabric_precision signature: (device_type: str) -> str
HEAD _select_compute_dtype("mps"): torch.float16
HEAD _select_fabric_precision("mps"): 16-mixed
trainer passes cfg.precision to dtype helper: False
trainer passes cfg.precision to fabric helper: False
```

## Impact

Makes fp32 an ineffective escape hatch for MPS numerical instability / AMP issues in Side-Step `training_v2`.

## Related

Related MPS training context, but not duplicates:

- #367: MPS mixed precision / FP16 gradients, closed unmerged
- #401: MPS training stabilization, merged
- #619: MPS LoRA training issues, closed

## Proposed fix

Thread requested precision through the helper functions and skip autocast when dtype is fp32.
